src/main/java/io/stargate/sgv2/jsonapi/service/operation/reranking/RerankingTask.java

```diff
@@ -349,4 +372,54 @@ public List<ScoredDocument> rerankedDocuments() {
       return rerankedDocuments;
     }
   }
+
+  // Inner class to handle metrics
+  private static class MetricsRecorder {
+    final String RERANK_INPUT_BYTES_METRICS = "reranking.input.bytes";
+    final String RERANKING_CALL_DURATION_METRICS = "reranking.call.duration";
+    final String RERANKING_PROVIDER_METRICS_TAG = "reranking.provider";
+    final String RERANKING_PROVIDER_MODEL_METRICS_TAG = "reranking.model";
+    final String TENANT_TAG = "tenant";
+    final String UNKNOWN_VALUE = "unknown";
+    private final MeterRegistry meterRegistry;
+    private final RerankingProvider rerankingProvider;
+    private final RequestContext requestContext;
+
+    public MetricsRecorder(
+        MeterRegistry meterRegistry,
+        RerankingProvider rerankingProvider,
+        RequestContext requestContext) {
+      this.meterRegistry = meterRegistry;
+      this.rerankingProvider = rerankingProvider;
+      this.requestContext = requestContext;
+    }
+
+    void recordInputSizes(List<String> passages) {
+      Objects.requireNonNull(passages);
+      DistributionSummary ds =
```

Comment:
Will dimensionality be a problem here, when we have break-down by tenant?
(maybe not just wondering since in the past we had some issues with certain metrics)